skopt/space.py

```diff
@@ -246,13 +269,27 @@ def __init__(self, categories, prior=None):
         self.categories = categories
         self.transformer = _CategoricalEncoder()
         self.transformer.fit(self.categories)
+        self.prior = prior
 
         if prior is None:
             prior = np.tile(1. / len(self.categories), len(self.categories))
 
         # XXX check that sum(prior) == 1
         self._rvs = rv_discrete(values=(range(len(self.categories)), prior))
 
+    def __eq__(self, other):
+        return (type(self) is type(other)
+                and self.categories == other.categories
+                and self.prior == other.prior)
+
+    def __repr__(self):
+        if len(self.categories) > 7:
```

Comment:
we show the first 3 and last 3; but if there are 7 I would just be replacing the middle element with `...` and taking up the same space. so I only truncate if there are more than 7. makes sense?